Index: freebsd-libs/lib/libdwarf/Makefile
===================================================================
--- freebsd-libs.orig/lib/libdwarf/Makefile
+++ freebsd-libs/lib/libdwarf/Makefile
@@ -1,6 +1,6 @@
 # $FreeBSD: stable/10/lib/libdwarf/Makefile 225227 2011-08-28 09:26:48Z kib $
 
-LIB=	dwarf
+LIB=	dwarf-freebsd
 
 SRCS=	\
 	dwarf_abbrev.c		\
@@ -20,7 +20,8 @@ SRCS=	\
 
 INCS=	dwarf.h libdwarf.h
 
-CFLAGS+=	-I${.CURDIR}
+LDADD+=		-L../libelf -lelf-freebsd
+CFLAGS+=	-I${.CURDIR} -I../libelf
 
 SHLIB_MAJOR=	3
 
