Index: freebsd-libs/lib/libgeom/Makefile
===================================================================
--- freebsd-libs.orig/lib/libgeom/Makefile
+++ freebsd-libs/lib/libgeom/Makefile
@@ -1,6 +1,7 @@
 # $FreeBSD: stable/10/lib/libgeom/Makefile 192376 2009-05-19 12:10:48Z trasz $
 
 LIB=	geom
+SHLIB_MAJOR= 1
 SHLIBDIR?= /lib
 SRCS+=	geom_getxml.c
 SRCS+=	geom_stats.c
@@ -16,6 +17,8 @@ WARNS?=	3
 DPADD=	${LIBBSDXML} ${LIBSBUF}
 LDADD=	-lbsdxml -lsbuf
 
+LDADD+=	-L../libsbuf/ -lrt -lbsd
+
 MAN=	libgeom.3
 
 MLINKS+= \
Index: freebsd-libs/lib/libutil/Makefile
===================================================================
--- freebsd-libs.orig/lib/libutil/Makefile
+++ freebsd-libs/lib/libutil/Makefile
@@ -5,15 +5,11 @@ SHLIBDIR?= /lib
 
 .include <bsd.own.mk>
 
-LIB=	util
+LIB=	util-freebsd
 SHLIB_MAJOR= 9
 
 SRCS=	_secure_path.c auth.c expand_number.c flopen.c fparseln.c gr_util.c \
-	hexdump.c humanize_number.c kinfo_getfile.c kinfo_getfile.c \
-	kinfo_getallproc.c kinfo_getproc.c kinfo_getvmmap.c kld.c \
-	login_auth.c login_cap.c \
-	login_class.c login_crypt.c login_ok.c login_times.c login_tty.c \
-	pidfile.c property.c pty.c pw_util.c quotafile.c realhostname.c \
+	hexdump.c humanize_number.c \
 	stub.c trimdomain.c uucplock.c
 INCS=	libutil.h login_cap.h
 
@@ -25,6 +21,14 @@ CFLAGS+= -DINET6
 
 CFLAGS+= -I${.CURDIR} -I${.CURDIR}/../libc/gen/
 
+OS!=	dpkg-architecture -qDEB_HOST_GNU_SYSTEM
+.if ${OS} == "kfreebsd-gnu"
+SRCS+=	login_cap.c login_class.c \
+	pidfile.c
+.endif
+
+LDADD= -lbsd -lfreebsd-glue
+
 MAN+=	expand_number.3 flopen.3 fparseln.3 hexdump.3 \
 	humanize_number.3 kinfo_getallproc.3 kinfo_getfile.3 \
 	kinfo_getproc.3 kinfo_getvmmap.3 kld.3 login_auth.3 login_cap.3 \
Index: freebsd-libs/lib/libkvm/Makefile
===================================================================
--- freebsd-libs.orig/lib/libkvm/Makefile
+++ freebsd-libs/lib/libkvm/Makefile
@@ -5,6 +5,7 @@ LIB=	kvm
 SHLIBDIR?= /lib
 SHLIB_MAJOR=	6
 CFLAGS+=-DLIBC_SCCS -I${.CURDIR}
+LDADD= -lbsd -lfreebsd-glue
 
 .if exists(${.CURDIR}/kvm_${MACHINE_ARCH}.c)
 KVM_ARCH=${MACHINE_ARCH}
Index: freebsd-libs/lib/libipx/Makefile
===================================================================
--- freebsd-libs.orig/lib/libipx/Makefile
+++ freebsd-libs/lib/libipx/Makefile
@@ -7,6 +7,10 @@ SRCS=	ipx_addr.c ipx_ntoa.c
 MAN=	ipx.3
 MLINKS+=ipx.3 ipx_addr.3 ipx.3 ipx_ntoa.3
 
+# for <netipx/ipx.h>
+CFLAGS+=	-I../../sys
+INCS+=		../../sys/netipx/ipx.h
+
 WARNS?=	2
 
 .include <bsd.lib.mk>
Index: freebsd-libs/lib/libelf/Makefile
===================================================================
--- freebsd-libs.orig/lib/libelf/Makefile
+++ freebsd-libs/lib/libelf/Makefile
@@ -61,7 +61,7 @@ INCS=	libelf.h gelf.h
 # problems when a header file in sys depends on a file in another
 # part of the tree, e.g. a machine dependent header.
 #
-SRCS+=	sys/elf32.h sys/elf64.h sys/elf_common.h
+#SRCS+=	sys/elf32.h sys/elf64.h sys/elf_common.h
 
 GENSRCS=	libelf_fsize.c libelf_msize.c libelf_convert.c
 CLEANFILES=	${GENSRCS}
