--- a/lib/libutil/Makefile
+++ b/lib/libutil/Makefile
@@ -5,18 +5,22 @@
 
 .include <bsd.own.mk>
 
-LIB=	util
+LIB=	util-freebsd
 SHLIB_MAJOR= 9
 
 SRCS=	_secure_path.c expand_number.c flopen.c fparseln.c gr_util.c \
-	hexdump.c humanize_number.c kinfo_getfile.c kinfo_getfile.c \
-	kinfo_getallproc.c kinfo_getproc.c kinfo_getvmmap.c kld.c \
-	login_auth.c login_cap.c \
-	login_class.c login_crypt.c login_ok.c login_times.c login_tty.c \
-	pidfile.c property.c pty.c pw_util.c quotafile.c realhostname.c \
+	hexdump.c humanize_number.c \
 	stub.c trimdomain.c uucplock.c
 INCS=	libutil.h login_cap.h
 
+SYS!=   dpkg-architecture -qDEB_HOST_GNU_SYSTEM
+.if ${SYS} == "kfreebsd-gnu"
+SRCS+=	login_cap.c login_class.c \
+	pidfile.c
+.endif
+
+LDADD=	-lbsd -lfreebsd-glue
+
 CFLAGS+= -DLIBC_SCCS
 
 .if ${MK_INET6_SUPPORT} != "no"
--- a/lib/libkvm/Makefile
+++ b/lib/libkvm/Makefile
@@ -4,6 +4,7 @@
 LIB=	kvm
 SHLIBDIR?= /lib
 CFLAGS+=-DLIBC_SCCS -I${.CURDIR}
+LDADD= -lbsd -lfreebsd-glue
 
 .if exists(${.CURDIR}/kvm_${MACHINE_ARCH}.c)
 KVM_ARCH=${MACHINE_ARCH}
