--- a/lib/libgeom/Makefile
+++ b/lib/libgeom/Makefile
@@ -1,6 +1,7 @@
 # $FreeBSD$
 
 LIB=	geom
+SHLIB_MAJOR= 1
 SHLIBDIR?= /lib
 SRCS+=	geom_getxml.c
 SRCS+=	geom_stats.c
@@ -16,6 +17,8 @@
 DPADD=	${LIBBSDXML} ${LIBSBUF}
 LDADD=	-lbsdxml -lsbuf
 
+LDADD+=	-L../libsbuf/ -lrt -lbsd
+
 MAN=	libgeom.3
 
 MLINKS+= \
--- a/lib/libutil/Makefile
+++ b/lib/libutil/Makefile
@@ -5,18 +5,22 @@
 
 .include <bsd.own.mk>
 
-LIB=	util
+LIB=	util-freebsd
 SHLIB_MAJOR= 9
 
 SRCS=	_secure_path.c expand_number.c flopen.c fparseln.c gr_util.c \
-	hexdump.c humanize_number.c kinfo_getfile.c kinfo_getfile.c \
-	kinfo_getallproc.c kinfo_getproc.c kinfo_getvmmap.c kld.c \
-	login_auth.c login_cap.c \
-	login_class.c login_crypt.c login_ok.c login_times.c login_tty.c \
-	pidfile.c property.c pty.c pw_util.c quotafile.c realhostname.c \
+	hexdump.c humanize_number.c \
 	stub.c trimdomain.c uucplock.c
 INCS=	libutil.h login_cap.h
 
+SYS!=   dpkg-architecture -qDEB_HOST_GNU_SYSTEM
+.if ${SYS} == "kfreebsd-gnu"
+SRCS+=	login_cap.c login_class.c \
+	pidfile.c
+.endif
+
+LDADD=	-lbsd -lfreebsd-glue
+
 CFLAGS+= -DLIBC_SCCS
 
 .if ${MK_INET6_SUPPORT} != "no"
--- a/lib/libkvm/Makefile
+++ b/lib/libkvm/Makefile
@@ -4,6 +4,7 @@
 LIB=	kvm
 SHLIBDIR?= /lib
 CFLAGS+=-DLIBC_SCCS -I${.CURDIR}
+LDADD= -lbsd -lfreebsd-glue
 
 .if exists(${.CURDIR}/kvm_${MACHINE_ARCH}.c)
 KVM_ARCH=${MACHINE_ARCH}
--- a/lib/libipx/Makefile
+++ b/lib/libipx/Makefile
@@ -7,6 +7,10 @@
 MAN=	ipx.3
 MLINKS+=ipx.3 ipx_addr.3 ipx.3 ipx_ntoa.3
 
+# for <netipx/ipx.h>
+CFLAGS+=	-I../../sys
+INCS+=		../../sys/netipx/ipx.h
+
 WARNS?=	2
 
 .include <bsd.lib.mk>
