--- a/lib/libutil/libutil.h
+++ b/lib/libutil/libutil.h
@@ -94,7 +94,7 @@
 int	extattr_string_to_namespace(const char *_string, int *_attrnamespace);
 int	flopen(const char *_path, int _flags, ...);
 int	forkpty(int *_amaster, char *_name,
-	    struct termios *_termp, struct winsize *_winp);
+	    const struct termios *_termp, const struct winsize *_winp);
 void	hexdump(const void *_ptr, int _length, const char *_hdr, int _flags);
 int	humanize_number(char *_buf, size_t _len, int64_t _number,
 	    const char *_suffix, int _scale, int _flags);
@@ -112,7 +112,7 @@
 int	kld_load(const char *_name);
 int	login_tty(int _fd);
 int	openpty(int *_amaster, int *_aslave, char *_name,
-	    struct termios *_termp, struct winsize *_winp);
+	    const struct termios *_termp, const struct winsize *_winp);
 int	pidfile_close(struct pidfh *_pfh);
 int	pidfile_fileno(const struct pidfh *_pfh);
 struct pidfh *
