--- a/sys/kern/subr_prf.c
+++ b/sys/kern/subr_prf.c
@@ -60,22 +60,22 @@
 #include <sys/syslog.h>
 #include <sys/cons.h>
 #include <sys/uio.h>
-#endif
 #include <sys/ctype.h>
+#endif
 #include <sys/sbuf.h>
 
 #ifdef DDB
 #include <ddb/ddb.h>
 #endif
 
+#ifdef _KERNEL
+
 /*
  * Note that stdarg.h and the ANSI style va_start macro is used for both
  * ANSI and traditional C compilers.
  */
 #include <machine/stdarg.h>
 
-#ifdef _KERNEL
-
 #define TOCONS	0x01
 #define TOTTY	0x02
 #define TOLOG	0x04
